src/mongo/s/commands/cluster_count_cmd.cpp

```diff
@@ -71,29 +71,19 @@ class ClusterCountCmd : public ErrmsgCommandDeprecated {
         out->push_back(Privilege(parseResourcePattern(dbname, cmdObj), actions));
     }
 
-    bool errmsgRun(OperationContext* opCtx,
-                   const std::string& dbname,
-                   const BSONObj& cmdObj,
-                   std::string& errmsg,
-                   BSONObjBuilder& result) override {
-        const NamespaceString nss(parseNs(dbname, cmdObj));
-        uassert(ErrorCodes::InvalidNamespace,
-                str::stream() << "Invalid namespace specified '" << nss.ns() << "'",
-                nss.isValid());
-
-        long long skip = 0;
+    bool run(OperationContext* opCtx,
+             const std::string& dbname,
+             const BSONObj& cmdObj,
+             std::string& errmsg,
+             BSONObjBuilder& result) override {
 
-        if (cmdObj["skip"].isNumber()) {
-            skip = cmdObj["skip"].numberLong();
-            if (skip < 0) {
-                errmsg = "skip value is negative in count query";
-                return false;
-            }
-        } else if (cmdObj["skip"].ok()) {
-            errmsg = "skip value is not a valid number";
-            return false;
+        const bool isExplain = true;
+        auto request = CountRequest::parseFromBSON(dbname, cmdObj, isExplain);
+        if (!request.isOK()) {
+            return request.getStatus();
         }
 
+
```

Comment:
Commenting here because I can't comment on lines not changed in the diff...

There is still a lot of parsing code below for the "query", "collation", "limit", and "skip" fields that hasn't been removed. This means that we are currently double-parsing the request: first with `CountRequest::parseFromBSON()`, and next with `cmdObj["query"]`, `bsonExtractTyedField(cmdObj, "collation", ...)`, etc.